### PR TITLE
[sw/silicon_creator] Track boot progress in boot_log

### PR DESCRIPTION
**Pull request by milesdai**
This PR introduces the `boot_log_t` struct that tracks information about the chip and the boot process. The `boot_log` is updated during the ROM and ROM_EXT boot flows.

### CODE REVIEW

File: sw/device/silicon_creator/lib/boot_log.h [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+extern "C" {
+#endif
+
+typedef struct boot_log {
+  hmac_digest_t digest;
+  uint32_t identifier;
+  chip_info_scm_revision_t chip_version;
+  uint32_t rom_ext_slot;
+  uint32_t bl0_slot;
+} boot_log_t;
```
Comment (milesdai) [inline sw/device/silicon_creator/lib/boot_log.h:25]: In the original spec, I think we also wanted a ROM version here. Is there a different version number besides the scm_revision?

File: sw/device/silicon_creator/lib/error.h [hunk 2]
```diff
@@ -152,7 +153,8 @@ enum module_ {
   X(kErrorRomExtBootFailed,           ERROR_(1, kModuleRomExt, kFailedPrecondition)), \
   \
   /* The high-byte of kErrorInterrupt is modified with the interrupt cause */ \
-  X(kErrorRomExtInterrupt,            ERROR_(0, kModuleRomExtInterrupt, kUnknown))
+  X(kErrorRomExtInterrupt,            ERROR_(0, kModuleRomExtInterrupt, kUnknown)), \
+  X(kErrorBootLogInvalid,             ERROR_(1, kModuleBootLog, kInternal))
```
Comment (alphan) [inline sw/device/silicon_creator/lib/error.h:157]: nit: empty line between different modules
```suggestion
  X(kErrorRomExtInterrupt,            ERROR_(0, kModuleRomExtInterrupt, kUnknown)), \
  \
  X(kErrorBootLogInvalid,             ERROR_(1, kModuleBootLog, kInternal))
```

**Review by alphan — APPROVED** [review]

**Comment by HU90m** [comment]
Is there a document where I can read up on the boot log?

**Comment by milesdai** [comment]
Sorry for the delayed response. Here's a [doc](https://docs.google.com/document/d/1jM_vVeKK4QryLWnTl9OlnuAUjj5zgx-S2UfwL-xCI_E/edit?usp=sharing) outlining the `boot_log`. I'll look into converting this into markdown.

**Pull request merged**